web/lib/mastermind.js

```diff
@@ -0,0 +1,25 @@
+const { eachSeries } = require('async');
+const { partialRight } = require('lodash');
+const Boom = require('boom');
+
+const stepHandler = (step, cb, req, reply) => step(req, reply, cb);
+
+const errorHandler = (err, reply) => {
+  if (err) return reply(Boom.boomify(err));
```

Comment:
nearly tempted if we have access to it (not sure) to add a either a log (to detect "impossible" scenarios) or a reply.continue

Reply:
Not quite sure I get what you mean here.

Reply:
non-erronous final callback should not get there